python/ray/tune/result_grid.py

```diff
@@ -69,6 +69,13 @@ def __init__(
         experiment_analysis: ExperimentAnalysis,
     ):
         self._experiment_analysis = experiment_analysis
+        self._results = list()
+
+        if experiment_analysis:
```

Comment:
`experiment_analysis` is a non-optional parameter, so we can just always do the thing in the if statement.